Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/view/ThumbnailImageView.kt

```diff
@@ -0,0 +1,11 @@
+package com.github.adamantcheese.chan.ui.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.ImageView
+
+class ThumbnailImageView(
```

Comment:
No, I need a separate class that extends from ImageView so I can tell it apart from the other ImageView because we already have an ImageView that is being used for video controlls.

Reply:
In other words findThumbnailImageView() won't work because we will have two ImageViews and I only need to find one (the thumbnail)

Reply:
https://github.com/Adamantcheese/Kuroba/pull/491/files/d21bec5be8b2933dd27540d8478bc3ac830f3f73#diff-24702c3f2446edd364337cb6657b47e5R150

Reply:
You can check to see if it's the playView ImageView instance in your search method and continue searching if it is. You do not need a separate class.

Reply:
That sounds kinda unnatural. I think that using a separate class is kinda more elegant way to deal with that.

Reply:
I'll rip it out myself then. I don't want ANOTHER thumbnail view class when we already have three of them.

Reply:
Alright, then how about I create a new class for video overlay instead? Like VideoOverlayImageView?

Reply:
Consider looking at PostImageThumbnailView before you start duplicating stuff again.

Reply:
I should have probably just named it differently. But it's hard to come up with new names. So what about extracting the view overlay into it's own class?

Reply:
PostImageThumbnailView contains some custom logic. I don't thing we want getting image url by longtapping it.

Reply:
Then maybe make it more generic? Add a flag to the constructor that would prevent that?

Reply:
I just wanted to add "let's not add new flags to it" but I was too late, damn. See, it already has PostXXX in it's name, so it's not suitable for the gallery.

Reply:
And we already use ThumbnailView for pins.

Reply:
And they all have custom logic that is only suitable for the place they are being used in right now. Trying to reuse them in different places is just not natural. The best thing to do here is to just create a new class for gallery thumbnails. No flags needed, no additional checks that will break some time in the future. If you ever want to add something new to the gallery thumbnail it's super simple because it lives in it's own class, no need to extract it into a separate one or adding even more checks.

Reply:
Then I would recommend moving the playIcon stuff into that thumbnail view, so that it has some use other than just being an empty wrapper class.

Reply:
You mean playIcon in PostImageThumbnailView? Then FixedRatioThumbnailView, which is used for pins  will also have playIcon and we probably don't want to see that icon on top of the pins' images.

Reply:
No, I mean move the playIcon stuff from MultiImageView into your new class. The end result should be similar to PostImageThumbnailView, but without the custom functionality, which is why I recommended looking at it in the first place.

Reply:
I think we shouldn't use ThumbnailImage in gallery whatsoever. ThumbnailImage is used alogside the ImageLoader (it uses Volley's callbacks) and we don't use Volley in gallery at all.

Reply:
Good sweet lord how am I not explaining this correctly. I'll just do it myself.

Reply:
Oh, I get it now. You meant to create a new class just with the playIcon. Kinda like what I meant when suggesting extracting the video overlay into it's own class. Yeah, that sounds reasonable.

Reply:
But we won't be able to use it with PostImageThumbnailView anyway since it already inherits from ThumbnailView which has tons of custom logic. So we will still have separate two classes with playIcon.

Reply:
>I'll just do it myself.

Please, just make sure that all findXXX() methods in MultiImageView can only find one class each because everything will break otherwise.

Reply:
Yes, that's what I meant. And I know that they should only find what they're looking for, which is why I suggested the nice and simple check instead of a refactor.

Reply:
> the nice and simple check

Uhhh.

Reply:
It's not nice and we would definitely forget about it in the future. Flags and checks are bad. And shared mutable state too.